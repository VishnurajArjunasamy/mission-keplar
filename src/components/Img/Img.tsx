import { FC } from "react";
import styles from "./Img.module.scss";
import emptyImage from "../../assets/images/empty-image.png";

interface ImgProps {
  src: string;
}

const Img: FC<ImgProps> = ({ src, ...props }) => {
  return (
    <img
      src={src}
      alt={"Movie"}
      className={styles["img-style"]}
      onError={(e) => {
        e.currentTarget.src = emptyImage;
        console.log("err");
      }}
      {...props}
    />
  );
};

export default Img;
