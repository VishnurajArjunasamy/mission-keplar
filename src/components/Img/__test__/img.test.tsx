import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Img from "../img";

describe("Img Component", () => {
  it("renders correctly with provided src and alt", () => {
    const { asFragment } = render(
      <Img src="https://example.com/image.jpg" alt="Test Image" />
    );
    const img = screen.getByAltText("Test Image");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "https://example.com/image.jpg");
    expect(asFragment()).toMatchSnapshot();
  });

  it("falls back to emptyUser image for user images on error", () => {
    const { asFragment } = render(
      <Img src="invalid-src" alt="User Image" isUser={true} />
    );
    const img = screen.getByAltText("User Image");

    // Simulate image load error
    fireEvent.error(img);

    expect(img).toHaveAttribute("src", expect.stringContaining("test-file-stub"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("falls back to placeholder blog image for non-user images on error", () => {
    const { asFragment } = render(
      <Img src="invalid-src" alt="Blog Image" isUser={false} />
    );
    const img = screen.getByAltText("Blog Image");

    // Simulate image load error
    fireEvent.error(img);

    expect(img).toHaveAttribute("src", expect.stringContaining("test-file-stub"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("uses default alt text when alt is not provided", () => {
    const { asFragment } = render(
      <Img src="https://example.com/image.jpg" />
    );
    const img = screen.getByAltText("image");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", "https://example.com/image.jpg");
    expect(asFragment()).toMatchSnapshot();
  });
});
