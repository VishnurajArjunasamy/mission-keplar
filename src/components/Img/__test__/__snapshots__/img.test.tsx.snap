// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img Component falls back to emptyUser image for user images on error 1`] = `
<DocumentFragment>
  <img
    alt="User Image"
    class="img"
    src="test-file-stub"
  />
</DocumentFragment>
`;

exports[`Img Component falls back to placeholder blog image for non-user images on error 1`] = `
<DocumentFragment>
  <img
    alt="Blog Image"
    class="img"
    src="test-file-stub"
  />
</DocumentFragment>
`;

exports[`Img Component renders correctly with provided src and alt 1`] = `
<DocumentFragment>
  <img
    alt="Test Image"
    class="img"
    src="https://example.com/image.jpg"
  />
</DocumentFragment>
`;

exports[`Img Component uses default alt text when alt is not provided 1`] = `
<DocumentFragment>
  <img
    alt="image"
    class="img"
    src="https://example.com/image.jpg"
  />
</DocumentFragment>
`;
